src/os400qc3.c

```diff
@@ -1079,7 +1077,7 @@ int _libssh2_hmac_update(libssh2_hmac_ctx *ctx,
 
     ctx->hash.Final_Op_Flag = Qc3_Continue;
     set_EC_length(errcode, sizeof(errcode));
-    Qc3CalculateHMAC((char *) data, &len, Qc3_Data, (char *) &ctx->hash,
+    Qc3CalculateHMAC((char *) data, &datalen, Qc3_Data, (char *) &ctx->hash,
```

Comment:
No, I already tried it. It causes a type mismatch error .-(

Reply:
Indeed! It's also one of those places where we must truncate the size (from `size_t` to `int` here). Strictly speaking these should be checked and bailing out or making an internal loop I think. (Won't be tackling these in this commit though.)